tests/io/dask/dataframe/test_repartition.py

```diff
@@ -0,0 +1,131 @@
+import pandas as pd
+import pytest
+
+from kartothek.core.factory import DatasetFactory  # noqa: E402
+from kartothek.io.dask.dataframe import read_dataset_as_ddf
+from kartothek.io.dask.repartition import (
+    calculate_repartition_ratio,
+    optimize,
+    repartition_ddf,
+)
+from kartothek.io.iter import store_dataframes_as_dataset__iter
+from kartothek.io_components.metapartition import SINGLE_TABLE
+
+
+@pytest.fixture
+def dataset_factory_function_scope(store_factory):
+    return DatasetFactory(dataset_uuid="test", store_factory=store_factory)
+
+
+@pytest.mark.parametrize("frac", (1.0, 0.5, 0.1))
+@pytest.mark.parametrize(
+    "df,target_nrows_per_partition",
+    (  # repartition by decreasing partition size, increasing number of partitions
+        (pd.DataFrame({"A": range(100), "part": ["A", "B", "C", "D"] * 25}), 100),
+        (pd.DataFrame({"A": range(2 ** 7), "part": ["A", "B", "C", "D"] * 2 ** 5}), 8),
+        # repartition by increasing partition size, decreasing number of partitions
+        (pd.DataFrame({"A": range(50), "part": range(50)}), 10),
+        (pd.DataFrame({"A": range(10), "part": range(10)}), 10),
```

Comment:
Same here. is this any different to the case above?

Reply:
taken care of using minimal combinations.